pipelines/streaming/README.md

```diff
@@ -0,0 +1,130 @@
+The
+[streaming](https://github.com/google/fhir-data-pipes/tree/master/pipelines/streaming)
+directory contains code to continuously listen for changes to an underlying
+OpenMRS MySQL database using
+[Debezium](https://debezium.io/documentation/reference/1.2/connectors/mysql.html).
+
+The Debezium-based streaming mode provides real-time downstream consumption of
+incremental updates, even for operations that were performed outside the OpenMRS
+API, e.g., data cleaning, module operations, and data syncing/migration. It
+captures incremental updates from the MySQL database binlog then streams both
+FHIR and non-FHIR data for downstream consumption. It is not based on Hibernate
+Interceptor or Event Module; therefore, all events are captured from day 0 and
+can be used independently without the need for a batch pipeline. It also
+tolerates failures like application restarts or crashes, as the pipeline will
+resume from the last processed offset.
+
+Streaming mode works only with OpenMRS; support for other data sources is not
+planned.
+
+## Prerequisites
+
+- An [OpenMRS](https://openmrs.org) instance with the latest version of the
+  [FHIR2 Module](https://addons.openmrs.org/show/org.openmrs.module.openmrs-fhir2-module)
+  installed.
+  - There is an
+    [OpenMRS Reference Application](https://github.com/google/fhir-data-pipes/wiki/Try-the-pipelines-using-local-test-servers)
+    image with these prerequisites and demo data you can use to try things out.
+- A target output for the data. Supported options are Apache Parquet files or a
+  FHIR server such as HAPI FHIR or
+  [Google Cloud Platform FHIR stores](https://cloud.google.com/healthcare/docs/how-tos/fhir).
+  - You can use our [HAPI FHIR server](#run-hapi-fhir-server-using-docker) image
+    for testing FHIR API targets.
+  - [Learn how to create a compatible GCP FHIR store](https://github.com/google/fhir-data-pipes/wiki/Create-a-Google-Cloud-FHIR-Store-and-BigQuery-Dataset),
+    if you want to use this option.
+
+## Getting Started
+
+1.  [Clone](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository)
+    the [FHIR Data Pipes](https://github.com/google/fhir-data-pipes) project to
+    your machine.
+1.  Set the `utils` directory to world-readable: `chmod -R 755 ./utils`.
```

Comment:
Is this still needed in non-docker mode?

Reply:
I'm not sure; this was in the original non-docker instructions.